app/home/routes.py

```diff
@@ -70,20 +72,27 @@ def get_segment(request):
 @blueprint.route("/property/create", methods=["GET", "POST"])
 @login_required
 def create_property():
+    from app.tasks import image_process
+
     form = CreatePropertyForm()
 
     if form.validate_on_submit():
         img_files = request.files.getlist("prop_photos")
-        temp_dir, temp_file_list = save_images_to_temp_folder(img_files)
+        folder_name = save_to_redis(img_files, current_user.username)
 
-        images_list = property_image_handler(temp_dir, temp_file_list)
-        img_list_to_json = json.dumps(images_list)
+        image_process.delay(folder_name, s3_prop_img_dir)
+        image_names = redis_client.hgetall(folder_name)
+        image_list = [
+            f"{image_name.decode('utf-8')}" for image_name in image_names.keys()
```

Comment:
am assuming folder name is the key and image names are a json value to that key
or am I wrongly looking at it
how are you doing it because it feels not right for you to get image by keys when we are supposed to have just 1 key holding a value of json image list

Reply:
The dictionary item looks something like this:
```
{"folder_name": {
    "image_name": image2_value(in_bytes),
    "image2_name": image2_value(in bytes)
}
```
I think this is convenient since we put everything in one object pushed it to Redis, and retain the key `folder_name` which is used by the worker by requesting `redis_client.hgetall(folder_name)` to get the image data and process it.
But perhaps you meant something like 
```
{"folder_name": ["image_name", "image2_name", image3_name"]}
``` 
in this case then the code should look like this
```
   folder_name = generate_dir_name(username)
   img_dict = {folder_name: []}
    for file in image_file_list:
        _, file_ext = os.path.splitext(file.filename)  # Get file extension
        new_img_name = uuid.uuid4().__str__()[:8]
        img_to_bytes = np.array(np.frombuffer(file.read(), np.uint8)).tobytes()
        folder_name[folder_name].append(f"{new_img_name}{file_ext}")
        redis_client.set(f"{new_img_name}{file_ext}")
    return folder_name
```
which could also work but both approaches achieve the same thing.